File: 프로그래머스/Programmers178870.java
import java.util.*;
class Solution {
    public int[] solution(int[] sequence, int k) {
        int start = 0, end = 0, sum = 0;
        int len = sequence.length;
        int minLen = Integer.MAX_VALUE;
        int[] answer = new int[2];

        while (true) {
            if (sum >= k) { // 합이 k 이상인 경우
                if (end - start < minLen) { // 최소 길이 갱신
                    minLen = end - start;
                    answer[0] = start;
                    answer[1] = end - 1;
                }
                sum -= sequence[start++]; // start 포인터를 오른쪽으로 이동
            } else if (end == len) { // end 포인터가 배열 끝에 도달한 경우
                break;
            } else { // 합이 k 미만인 경우
                sum += sequence[end++]; // end 포인터를 오른쪽으로 이동
            }
        }

        return answer;
    }
}

File: 프로그래머스/Programmers178870.js
function solution(sequence, k) {
    var answer = [];
    let sum = 0; // 초기 합을 0으로 설정
    let left = 0;
    let minLength = Number.MAX_VALUE;
    let start = -1; // 시작 인덱스
    let end = -1; // 끝 인덱스

    for (let right = 0; right < sequence.length; right++) {
        sum += sequence[right];

        // sum이 k보다 커질 때까지 left 인덱스를 증가
        while (sum > k) {
            sum -= sequence[left];
            left++;
        }

        // sum이 k와 같을 때
        if (sum === k) {
            let length = right - left + 1; // 부분 수열의 길이
            if (length < minLength) {
                minLength = length;
                start = left; // 시작 인덱스 저장
                end = right; // 끝 인덱스 저장
            }
        }
    }

    return [start, end]; // 최종 결과 반환
}
